### PR TITLE
ERR: stringify error message from parsers.pyx, closes #29233

### LINKED ISSUE #29233: read_csv encode operation on source
#### Code Sample, a copy-pastable example if possible

```python
>>> import pandas as pd
>>> df = pd.read_csv("historical_dataset.csv")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/pandas/io/parsers.py", line 709, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/usr/lib/python3/dist-packages/pandas/io/parsers.py", line 449, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/usr/lib/python3/dist-packages/pandas/io/parsers.py", line 818, in __init__
    self._make_engine(self.engine)
  File "/usr/lib/python3/dist-packages/pandas/io/parsers.py", line 1049, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/usr/lib/python3/dist-packages/pandas/io/parsers.py", line 1695, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 402, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 718, in pandas._libs.parsers.TextReader._setup_parser_source
FileNotFoundError: File b'historical_dataset.csv' does not exist
```
#### Problem description

Noticed that when source does not exist, there is a leading 'b' on the source on the error message. For Python3, is it necessary to do the encode operation on line 667 of https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/parsers.pyx? Works fine for Python2.

#### Output of ``pd.show_versions()``
<details>
```
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.0.0-31-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.1
numpy            : 1.14.5
pytz             : 2018.5
dateutil         : 2.8.0
pip              : 19.2.3
setuptools       : 41.2.0
Cython           : 0.29.13
pytest           : None
hypothesis       : None
sphinx           : 1.7.6
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.2.3
html5lib         : None
pymysql          : 0.9.3
psycopg2         : None
jinja2           : 2.10
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.2.3
matplotlib       : 2.2.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.1.0
sqlalchemy       : 1.3.8
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
```
</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #29233
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/parser/test_common.py [hunk 1]
```diff
@@ -2160,6 +2160,14 @@ def test_suppress_error_output(all_parsers, capsys):
     assert captured.err == ""
 
 
+def test_err_msg_bytes():
+    # GH#29233
+    try:
+        pd.read_csv("nonexistent_name")
+    except FileNotFoundError as err:
+        assert err.strerror == "File nonexistent_name does not exist"
```
Comment (simonjayhawkins) [inline pandas/tests/io/parser/test_common.py]: can pytest.raises not be used?

Comment (jbrockmendel) [reply pandas/tests/io/parser/test_common.py]: good idea.  i think this was split off from a branch where i was also checking err.errno

Comment (jbrockmendel) [reply pandas/tests/io/parser/test_common.py]: also looks like theres another test i can merge this into

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**